internal/controlplane/identity_events.go

```diff
@@ -167,30 +172,24 @@ func DeleteUser(ctx context.Context, store db.Store, authzClient authz.Client, u
 		return fmt.Errorf("error getting projects for user %v", err)
 	}
 
+	l.Debug().Msg("deleting user from authorization system")
 	// We delete the user from the authorization system first
 	if err := authzClient.DeleteUser(ctx, userId); err != nil {
 		return fmt.Errorf("error deleting authorization tuple %v", err)
 	}
 
 	// We only delete the user if it still exists in the database
 	if userDBID != nil {
+		l.Debug().Msg("deleting user from database")
 		if err := qtx.DeleteUser(ctx, *userDBID); err != nil {
 			return fmt.Errorf("error deleting user %v", err)
 		}
 	}
 
 	for _, proj := range projs {
-		// Given that we've deleted the user from the authorization system,
-		// we can now check if there are any role assignments for the project.
-		as, err := authzClient.AssignmentsToProject(ctx, proj)
```

Comment:
so this is also changing the logic to delete the project if there's still assignments?

Reply:
that was repeated logic that already existed in the CleanUnamangedProjects. We were repeating that.